lib/ansible/constants.py

```diff
@@ -176,11 +177,28 @@ def __getitem__(self, y):
     become_flags=('ansible_become_flags', ),
 )
 
-# POPULATE SETTINGS FROM CONFIG ###
+_myself = sys.modules[__name__]
 config = ConfigManager()
 
-# Generate constants from config
-for setting in config.data.get_settings():
+
+def __getattr__(name):
+
+    if name in _myself.__dict__:
+        value = getattr(_myself, name)
+    else:
+        setting = config.data.get_setting(name)
+        for warn in config.WARNINGS:
+            _warning(warn)
+        if setting:
+            value = _extract_value(setting)
+            setattr(_myself, name, value)
+        else:
+            raise ImportError("No constant nor setting named: %s" % name)
```

Comment:
Actually it needs to be `AttributeError` to keep everything happy (which the import machinery will turn into an `ImportError` if it was sourced by an import)